spark/src/test/scala/org/apache/spark/sql/delta/IdentityColumnSyncSuite.scala

```diff
@@ -0,0 +1,258 @@
+    withSimpleGeneratedByDefaultTable(startsWith = -10L, incrementBy = -2L) {
+      sql(s"INSERT INTO $tblName (id, value) VALUES (1, 'a'), (2, 'b'), (-9, 'c')")
+      sql(s"ALTER TABLE $tblName ALTER COLUMN id SYNC IDENTITY")
+      sql(s"INSERT INTO $tblName (value) VALUES ('d'), ('e'), ('f')")
+
+      val result = spark.read.table(tblName)
+        .as[IdentityColumnTestTableRow]
+        .collect()
+        .sortBy(_.id)
+      assert(result.length === 6)
+      assert(result.takeRight(3) === Seq(IdentityColumnTestTableRow(-9, "c"),
+                                         IdentityColumnTestTableRow(1, "a"),
+                                         IdentityColumnTestTableRow(2, "b")))
+      checkGeneratedIdentityValues(
+        sortedRows = result.take(3),
+        start = -10,
+        step = -2,
+        expectedLowerBound =
+          highWaterMark(DeltaLog.forTable(spark, TableIdentifier(tblName)).snapshot, "id"),
+        expectedUpperBound = -10,
+        expectedDistinctCount = 3)
+    }
+  }
+
+  test("alter table sync identity - deleting high watermark rows followed by sync identity" +
+    " brings down the highWatermark") {
+    for (generatedAsIdentityType <- GeneratedAsIdentityType.values) {
+      withTable(tblName) {
+        createTableWithIdColAndIntValueCol(tblName, generatedAsIdentityType, Some(1L), Some(10L))
+        val deltaLog = DeltaLog.forTable(spark, TableIdentifier(tblName))
+        (0 to 4).foreach { v =>
+          sql(s"INSERT INTO $tblName(value) VALUES ($v)")
+        }
+
+        checkAnswer(sql(s"SELECT max(id) FROM $tblName"), Row(41))
+        sql(s"DELETE FROM $tblName WHERE value IN (0, 3, 4)")
+        assert(highWaterMark(deltaLog.snapshot, "id") === 41L)
+        sql(s"ALTER TABLE $tblName ALTER COLUMN id SYNC IDENTITY")
+        assert(highWaterMark(deltaLog.snapshot, "id") === 21L)
+        sql(s"INSERT INTO $tblName(value) VALUES (8)")
+        checkAnswer(sql(s"SELECT max(id) FROM $tblName"), Row(31))
+        checkAnswer(sql(s"SELECT COUNT(DISTINCT id) == COUNT(*) FROM $tblName"), Row(true))
+      }
+    }
+  }
+
+  test("alter table sync identity overflow") {
+    withSimpleGeneratedByDefaultTable(startsWith = 1L, incrementBy = 10L) {
+      sql(s"INSERT INTO $tblName VALUES (${Long.MaxValue}, 'a')")
+      intercept[ArithmeticException](sql(s"ALTER TABLE $tblName ALTER COLUMN id SYNC IDENTITY"))
+    }
+  }
+
+  test("alter table sync identity non delta") {
+    withTable(tblName) {
+      sql(
+        s"""
+           |CREATE TABLE $tblName (
+           |  id BIGINT,
+           |  value INT
+           |) USING parquet;
+           |""".stripMargin)
+      val ex = intercept[AnalysisException] {
+        sql(s"ALTER TABLE $tblName ALTER COLUMN id SYNC IDENTITY")
+      }
+      assert(ex.getMessage.contains("ALTER TABLE ALTER COLUMN SYNC IDENTITY is only supported"))
+    }
+  }
+
+  test("alter table sync identity non identity column") {
```

Comment:
Same here, it would be great to easily know the intention of the test from the name.